File: perl/goforperlprogrammers/code/pi.pl
#!/usr/bin/perl
use strict;

my $n = 5000000;
my $delta = 1/$n;
my ($x, $sum);
for (my $i = 1; $i <= $n; $i++) {
    $x = ($i - 0.5) * $delta;
	$sum += 1.0 / (1.0 + $x*$x);
}
print 4 * $delta * $sum, "\n";

File: perl/goforperlprogrammers/code/pi.go
package main

import (
	"fmt"
)

func main() {
	n := 5000000
	delta := 1.0 / float64(n)
	var sum, x float64

	for i := 1; i <= n; i++ {
		x = (float64(i) - 0.5) * delta
		sum += 1.0 / (1.0 + x*x)
	}
	var pi = 4.0 * delta * sum
	fmt.Println("Pi is approximately", pi, "after", n, "iterations.")
}
